monai/data/image_writer.py

```diff
@@ -0,0 +1,725 @@
+            image_mode=kwargs.pop("image_mode", None),
+            scale=self.scale,  # type: ignore
+            **kwargs,
+        )
+        self.data_obj.save(filename_or_obj, **kwargs)
+
+    @classmethod
+    def get_meta_info(cls, metadata: Optional[Mapping] = None):
+        if not metadata:
+            return None
+        return metadata.get("spatial_shape")
+
+    @classmethod
+    def resample_and_clip(
+        cls,
+        data_array: NdarrayOrTensor,
+        output_spatial_shape: Optional[Sequence[int]] = None,
+        mode: Union[InterpolateMode, str] = InterpolateMode.BICUBIC,
+    ):
+        """
+        This method assumes the 'channel-last' format
+        """
+
+        data: np.ndarray = convert_data_type(data_array, np.ndarray)[0]  # type: ignore
+        if output_spatial_shape is not None:
+            output_spatial_shape_ = ensure_tuple_rep(output_spatial_shape, 2)
+            mode = look_up_option(mode, InterpolateMode)
+            align_corners = None if mode in (InterpolateMode.NEAREST, InterpolateMode.AREA) else False
+            xform = Resize(spatial_size=output_spatial_shape_, mode=mode, align_corners=align_corners)
+            _min, _max = np.min(data), np.max(data)
+            if len(data.shape) == 3:
+                data = np.moveaxis(data, -1, 0)  # to channel first
+                data = xform(data)  # type: ignore
+                data = np.moveaxis(data, 0, -1)
+            else:  # (H, W)
+                data = np.expand_dims(data, 0)  # make a channel
+                data = xform(data)[0]  # type: ignore
+            if mode != InterpolateMode.NEAREST:
+                data = np.clip(data, _min, _max)  # type: ignore
+        return data
+
+    @classmethod
+    def create_backend_obj(
+        cls,
+        data_array: NdarrayOrTensor,
+        dtype: DtypeLike = None,
+        scale: Optional[int] = 255,
+        reverse_indexing: bool = True,
+        **kwargs,
+    ):
+        data: np.ndarray = super().create_backend_obj(data_array)
+        if scale:
+            # scale the data to be in an integer range
+            data = np.clip(data, 0.0, 1.0)  # type: ignore # png writer only can scale data in range [0, 1]
+            if scale == np.iinfo(np.uint8).max:
+                data = (scale * data).astype(np.uint8, copy=False)
+            elif scale == np.iinfo(np.uint16).max:
+                data = (scale * data).astype(np.uint16, copy=False)
+            else:
+                raise ValueError(f"Unsupported scale: {scale}, available options are [255, 65535].")
+        if dtype is not None:
+            data = data.astype(dtype, copy=False)
+        if reverse_indexing:
+            data = np.moveaxis(data, 0, 1)
+
+        return PILImage.fromarray(data, mode=kwargs.pop("image_mode", None))
+
+
+def init():
+    """Initialize the image writer modules according to the file extension name."""
+    register_writer(".png", PILWriter)
```

Comment:
We can add ".jpg", ".bmp", ".tif", and others to use PILWriter. TIF files are sometimes used for 16-data other formats don't store well.